### PR TITLE
docs: Write Git documentation

### PR DESCRIPTION
**Pull request by chanbakjsd**
This PR adds Git documentation including a quick start guide and
suggested conventions for commit messages and branches.

[SCSE-24]

### CODE REVIEW

File: docs/code-organization/naming-convention.mdx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+This page contains naming conventions for commit messages and branches.
+
+## Commit Messages
+
+Commit message titles should be in the form `type: Short description of the
+change` or `type(subsystem): Short description of the change` as per
+[Conventional Commits][conventional-commit].
+
+Furthermore, the ticket number (if applicable) should be contained at the
```
Comment (RealDyllon) [inline docs/code-organization/naming-convention.mdx:15]: can we specify here: "the ticket number **from the linear task**" ?

File: docs/code-organization/git-quickstart.mdx [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+[git-add]: https://git-scm.com/docs/git-add
+[git-rm]: https://git-scm.com/docs/git-rm
+[git-commit]: https://git-scm.com/docs/git-commit
+[git-switch]: https://git-scm.com/docs/git-switch
+[git-pull]: https://git-scm.com/docs/git-pull
+[git-push]: https://git-scm.com/docs/git-push
+[git-reset]: https://git-scm.com/docs/git-reset
+
+[pro-git-chap-3]: https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell
+[create-github-repo]: https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-new-repository
```
Comment (RealDyllon) [inline docs/code-organization/git-quickstart.mdx:331]: Using link vars is a nice touch

File: docs/code-organization/naming-convention.mdx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+```
+feat(payment): Add Stripe checkout
+
+This commit adds Stripe integration to the payment page. The STRIPE_API_KEY
+environment variable should be set to the secret key available at Stripe
+dashboard.
+
+[SCSE-24]
+```
```
Comment (RealDyllon) [inline docs/code-organization/naming-convention.mdx:36]: feel free to reword the commit message in the example to contain more details (use positive examples rather than negative examples). 

We can, however, have a 'negative example' of a commit message - one that only contains poor content - in order to highlight multiple problems to the reader.

File: docs/code-organization/git-quickstart.mdx [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+dquote>
+dquote>[SCSE-24]"
+[main f253a79] docs: Fix typo in Example docs
+ Date: Thu Oct 13 23:39:33 2022 +0800
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+```
+
+:::caution
+Note that by doing this, you are recreating history!
+NEVER ever re-create history that is already public and in the `main` branch as
```
Comment (RealDyllon) [inline docs/code-organization/git-quickstart.mdx:152]: Can we reword this to say: don't amend commits on any branch that has been pushed, unless you know for certain that you are the only person working on it (or something to that effect)?

File: docs/code-organization/git-quickstart.mdx [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+## Branching
+
+You should definitely create a branch for what you are working on before you
+start working on your super cool feature.
+
+To quote [Pro Git Chapter 3][pro-git-chap-3], branching means you diverge from
+the main line of development and continue to do work without messing with
+that main line.
+
+The [`git switch`][git-switch] command has been added in a recent update of Git
```
Comment (RealDyllon) [inline docs/code-organization/git-quickstart.mdx:165]: let's specify that this is only available git v2.23+, and the fact that it's experimental, with api subject to change.

File: docs/code-organization/git-quickstart.mdx [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+remote:
+remote: Create a pull request for 'docs/SCSE-24' on GitHub by visiting:
+remote:      https://github.com/ntuscse/docs/pull/new/docs/SCSE-24
+remote:
+To github.com/ntuscse/docs
+ * [new branch]      docs/SCSE-24 -> docs/SCSE-24
+```
+
+If you forgot to create a branch, you can still specify what the name at GitHub
+should be. In the following example, we assume you worked on `main` and the
```
Comment (RealDyllon) [inline docs/code-organization/git-quickstart.mdx:219]: Let's be more verbose, by saying i.e. "for example, if you **accidentally** worked on the main branch locally, and wish to move your commits to your task branch xyz...", or something to that effect

File: docs/code-organization/git-quickstart.mdx [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+remote:      https://github.com/ntuscse/docs/pull/new/docs/SCSE-24
+remote:
+To github.com/ntuscse/docs
+ * [new branch]      main -> docs/SCSE-24
+```
+
+## What Next?
+
+This document is currently work in progress! If there is anything you are
+unclear about, ask about them in our chat and we will be using those valuable
```
Comment (RealDyllon) [inline docs/code-organization/git-quickstart.mdx:236]: change chat to discord

File: docs/code-organization/git-quickstart.mdx [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+
+Git is just a tool to help manage different versions of code as we collaborate
+and is nothing if we don't use it so start familiarizing yourself with it! You
+may wish to do so by [creating your own GitHub repository][create-github-repo]
+and playing around with it.
+
+You might want to check out the [naming conventions](naming-convention) we have
+written to help keep everything tidy.
+
+## Appendix A: Fixing History
```
Comment (RealDyllon) [inline docs/code-organization/git-quickstart.mdx:247]: good section 👍🏼

File: docs/code-organization/git-quickstart.mdx [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+in.
+
+#### Option 1: Force push (softer version)
+
+Perhaps you were just pushing to the branch of your PR and it's actually fine
+to override it.
+
+:::caution
+Before we continue, make extra sure that you are pushing to the branch of your
+PR and not someone else's branch. You can check this by reading the right side
```
Comment (RealDyllon) [inline docs/code-organization/git-quickstart.mdx:274]: reword to "make sure you are pushing the branch of your own task, and not someone elses.....", or something to that effect.

File: docs/code-organization/git-quickstart.mdx [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+---
+
+# Git Quickstart
+
+This page contains [Git][git-link] commands that may be useful
+during development. Examples have been provided where the command to be
+executed is prefixed with `$` and the sample output are not.
+
+Please refer to the [documentation for Git][git-doc] for a
+deeper understanding in Git.
```
Comment (RealDyllon) [inline docs/code-organization/git-quickstart.mdx:12]: let's add link to the git setup doc at the top: https://docs.ntuscse.com/docs/getting-started/github

**Review by RealDyllon — CHANGES_REQUESTED** [review]

**Comment by chanbakjsd** [comment]
I've made the changes as requested, please do review again. :)

**Comment by RealDyllon** [comment]
@chanbakjsd in future let's allow the thread creator to mark threads as resolved (as a final acknowledgement). This is important as branch protection prevents merging until all PR threads are resolved.

**Comment by chanbakjsd** [comment]
Understood, I believe the PR author can (and should) mark the trivial resolutions as acknowledgements however. Let's continue this discussion in Discord as it is not related to this PR directly.

**Review by RealDyllon — APPROVED** [review]
LGTM
